librz/core/cmd/cmd_debug.c

```diff
@@ -1066,34 +1066,15 @@ static void backtrace_vars(RzCore *core, RzList /*<RzDebugFrame *>*/ *frames) {
 		rz_reg_setv(r, bp, s);
 		rz_reg_setv(r, sp, b);
 		//////////
-		char flagdesc[1024], flagdesc2[1024];
-		RzFlagItem *fi = rz_flag_get_at(core->flags, f->addr, true);
-		flagdesc[0] = flagdesc2[0] = 0;
-		if (fi) {
-			if (fi->offset != f->addr) {
-				int delta = (int)(f->addr - fi->offset);
-				if (delta > 0) {
-					snprintf(flagdesc, sizeof(flagdesc),
-						"%s+%d", fi->name, delta);
-				} else if (delta < 0) {
-					snprintf(flagdesc, sizeof(flagdesc),
-						"%s%d", fi->name, delta);
-				} else {
-					snprintf(flagdesc, sizeof(flagdesc),
-						"%s", fi->name);
-				}
-			} else {
-				snprintf(flagdesc, sizeof(flagdesc),
-					"%s", fi->name);
-			}
-		}
+		char *flagdesc = rz_core_addr_get_flag_offset(core, f->addr);
```

Comment:
Please don't pass `core` here. Just pass the `RzFlag` instance. We want to reduce the dependencies on COre, not increase them.

Reply:
For this I would also suggest move the function into the `rz_flag` module.